src/Util/JSON.php

```diff
@@ -22,12 +22,16 @@ final class JSON
      * @param int   $options  Bitmask consisting of JSON_* constants
      * @param int   $maxDepth The maximum depth allowed for serializing $data
      *
-     * @return mixed
+     * @return false|string
```

Comment:
`false` should not be possible here...

Reply:
Why not? `"false"` is a valid JSON string and so also `boolean(false)` is a valid return value

Reply:
Then any other `scalar` is fair game...
Since we bumped PHP requirement, couldn't we drop this in favor of  `JSON_THROW_ON_ERROR` and call it a day?

Reply:
` JSON_THROW_ON_ERROR` has nothing to do with this. As I said, the word `false` is deserialized to a `boolean(false)` because it's valid, not because an error happened. Anyway, there is one thing that is actually wrong: this method should return indeed `mixed` as before, so I'm not sure why I changed this besides thinking that PHPStan was complaining or that I misread its output. I will check it again and revert the change if this turns out to be true

Reply:
The [PHP documentation](https://www.php.net/manual/en/function.json-encode.php) says that `false|string` is the return type for `json_encode()`, and I think that I did this change because PHPStan was complaining due to it not understanding the conditional check on `json_last_error()`

>Since we bumped PHP requirement, couldn't we drop this in favor of `JSON_THROW_ON_ERROR` and call it a day?

Indeed this or suppressing the PHPStan error are the proper ways to fix the issue. The problem with your suggestion is that that constant has been introduced in PHP `7.3`, while we still support PHP `7.2`, so the only thing left is suppressing the error for now